gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -98,17 +98,20 @@ QueryRequest buildQueryRequest(
     // 3. Add GroupBy
     addGroupByExpressions(builder, request);
 
-    // 4. If there's no Group By, Set Limit, Offset and Order By.
-    // Otherwise, specify a large limit and track actual limit, offset and order by expression list
-    // so we can compute
-    // these once the we get the results.
-    if (requestContext
-        .hasGroupBy()) { // Will need to do the Ordering, Limit and Offset ourselves after we get
-      // the Group By Results
-      builder.setLimit(DEFAULT_QUERY_SERVICE_GROUP_BY_LIMIT);
-      requestContext.setOrderByExpressions(getRequestOrderByExpressions(request));
-    } else { // No Group By: Use Pinot's Order By, Limit and Offset
-      addSortLimitAndOffset(request, builder);
+    // 4. Add Order By
+    addSort(request, builder);
+
+    // 5. Set Limit, Offset
+    if (requestContext.hasGroupBy()) {
+      // providing both group by and order by doesn't provide right results in pinot
```

Comment:
3 concerns:

1. Does this break support for the "other" flag. E.g. if I'm saying give me 5 groups + other, do we need to over-query? We should make sure that's well tested. I don't think other makes sense when using offset, so not concerned about that - just the basic case of limit + other.
2. performance - forcing a query of limit+offset can easily run into exhaustion issues, moving a lot of data. At a minimum we should update our validation in this case that limit + offset is less than max limit, but I'd concern whether we really even want to support offset on a group by given the first concern. We could just throw if the two are used together.
3. In this function, we're applying offset in the db in QS in some cases, in others we're skipping. In the result post process function that this handler implements, we always skip (assume that it was not applied in QS). I don't have the code up outside the diff so not sure if that path is only hit in the group by case, but even if so, those types of assumptions make the code exceedingly error prone and complex to follow.

Reply:
1. Validated the `other` flag, and it works fine with this change
2. Updated such that group by and offset results in unsupported operation exception
3. No longer an issue, given offset + group by are not supported

BTW I have done some testing on saas dev sandbox. How do we exhaustively test all possible scenarios?

Reply:
All scenarios is hard :) I'd probably test a matrix of [different permutations of explorer params - time series, group by, group by w/ rest, grouped time series, grouped time series w/ rest, ordered group by etc.] x [different data sources - ES, QS]

Reply:
Made all required code changes and fixed the tests as well. Please have a look.